ci/environment-3.10.yaml

```diff
@@ -9,6 +9,7 @@ dependencies:
   - pandas
   - pyarrow
   - pytest
+  - gcsfs
   - grpcio<1.45
```

Comment:
We had this pin due to a seg fault on windows, but it's been a year since this happen can we try to unpin this and see what happen? I understand it's not part of this PR, we can leave it for a separate PR.

Reply:
I'll try in a separate PR. Thanks for the tip!